Tests/DatadogTests/Datadog/Core/Upload/DataUploaderTests.swift

```diff
@@ -7,26 +7,62 @@
+        XCTAssertEqual(validURL1.url, URL(string: "https://api.example.com/v1/endpoint/abc?ddsource=mobile&batch_time=1576404000000"))
+
+        dateProvider.advance(bySeconds: 9.999)
 
-        let validURL2 = try DataUploadURL(endpointURL: "https://api.example.com/v1/endpoint/", clientToken: "abc")
-        XCTAssertEqual(validURL2.url, URL(string: "https://api.example.com/v1/endpoint/abc?ddsource=mobile"))
+        let validURL2 = try UploadURLProvider(endpointURL: "https://api.example.com/v1/endpoint/", clientToken: "abc", dateProvider: dateProvider)
+        XCTAssertEqual(validURL2.url, URL(string: "https://api.example.com/v1/endpoint/abc?ddsource=mobile&batch_time=1576404009999"))
     }
 
     func testWhenClientTokenIsInvalid_itThrowsProgrammerError() {
-        XCTAssertThrowsError(try DataUploadURL(endpointURL: "https://api.example.com/v1/endpoint", clientToken: "")) { error in
+        XCTAssertThrowsError(try UploadURLProvider(endpointURL: "https://api.example.com/v1/endpoint", clientToken: "", dateProvider: dateProvider)) { error in
             XCTAssertEqual((error as? ProgrammerError)?.description, "Datadog SDK usage error: `clientToken` cannot be empty.")
         }
     }
 
     func testWhenEndpointURLIsInvalid_itThrowsProgrammerError() {
-        XCTAssertThrowsError(try DataUploadURL(endpointURL: "", clientToken: "abc")) { error in
+        XCTAssertThrowsError(try UploadURLProvider(endpointURL: "", clientToken: "abc", dateProvider: dateProvider)) { error in
             XCTAssertEqual((error as? ProgrammerError)?.description, "Datadog SDK usage error: `endpointURL` cannot be empty.")
         }
     }
+
+    // MARK: - Logs endpoint
+
+    func testUSLogsEndpoint() {
+        let urlUS = Datadog.Configuration.LogsEndpoint.us.url
+        let urlProvider = try! UploadURLProvider(
+            endpointURL: urlUS,
+            clientToken: "abc",
+            dateProvider: dateProvider
+        )
+        XCTAssertEqual(urlProvider.url, URL(string: "https://mobile-http-intake.logs.datadoghq.com/v1/input/abc?ddsource=mobile&batch_time=1576404000000"))
+    }
+
+    func testEULogsEndpoint() {
+        let urlEU = Datadog.Configuration.LogsEndpoint.eu.url
+        let urlProvider = try! UploadURLProvider(
+            endpointURL: urlEU,
+            clientToken: "abc",
+            dateProvider: dateProvider
+        )
+        XCTAssertEqual(urlProvider.url, URL(string: "https://mobile-http-intake.logs.datadoghq.eu/v1/input/abc?ddsource=mobile&batch_time=1576404000000"))
+    }
+
+    func testCustomLogsEndpoint() {
+        let urlCustom = Datadog.Configuration.LogsEndpoint.custom(url: "foo.bar").url
+        let urlProvider = try! UploadURLProvider(
+            endpointURL: urlCustom,
+            clientToken: "abc",
+            dateProvider: dateProvider
+        )
+        XCTAssertEqual(urlProvider.url, URL(string: "foo.bar/abc?ddsource=mobile&batch_time=1576404000000"))
+    }
```

Comment:
Those tests do exactly the same assertion as `func testItBuildsValidURL()` - _"When a given URL string is provided, `UploadURLProvider` decorates it with expected URL queries"_. Asserting values of the URL for different endpoints (`.us`, `.eu`, `.custom(...)`) is a better fit for configuration builder tests (`DatadogConfigurationTests`) as we also need to assert the defaults it produces (when no endpoint is set explicitly).

In other words:
* in `DataUploaderTests` we can test: 
   * `.us` endpoint string
   * `.eu` endpoint string
   * `.custom` endpoint string
* in `DatadogConfigurationTests ` we could test: 
   * `.us` endpoint string
   * `.eu` endpoint string
   * `.custom` endpoint string
   * default endpoint if not explicitly set by the user 👈 


```suggestion
```

Reply:
done ✅